newIDE/app/src/EventsSheet/ParameterFields/ExternalLayoutNameField.js

```diff
@@ -7,35 +7,127 @@ import {
   type ParameterFieldInterface,
   type FieldFocusFunction,
 } from './ParameterFieldCommons';
-import { type ExpressionAutocompletion } from '../../ExpressionAutocompletion';
+import Functions from '@material-ui/icons/Functions';
+import SelectField, { type SelectFieldInterface } from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+import FlatButton from '../../UI/FlatButton';
+import TypeCursorSelect from '../../UI/CustomSvgIcons/TypeCursorSelect';
+import { Trans, t } from '@lingui/macro';
+import RaisedButton from '../../UI/RaisedButton';
+import { TextFieldWithButtonLayout } from '../../UI/Layout';
 
 export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
   function ExternalLayoutNameField(props: ParameterFieldProps, ref) {
-    const field = React.useRef<?GenericExpressionField>(null);
+    const field = React.useRef<?(
+      | GenericExpressionField
+      | SelectFieldInterface
+    )>(null);
     const focus: FieldFocusFunction = options => {
       if (field.current) field.current.focus(options);
     };
     React.useImperativeHandle(ref, () => ({
       focus,
     }));
 
-    const externalLayoutNames: Array<ExpressionAutocompletion> = props.project
-      ? enumerateExternalLayouts(props.project).map(externalLayout => ({
-          kind: 'Text',
-          completion: `"${externalLayout.getName()}"`,
-        }))
+    const externalLayoutNames = props.project
+      ? enumerateExternalLayouts(props.project).map(externalLayout =>
+          externalLayout.getName()
+        )
       : [];
+    const isCurrentValueInLayoutsList = !!externalLayoutNames.find(
+      layoutName => `"${layoutName}"` === props.value
+    );
+
+    // If the current value is not in the list of layouts, display an expression field.
+    const [isExpressionField, setIsExpressionField] = React.useState(
+      !!props.value && !isCurrentValueInLayoutsList
+    );
+
+    const switchFieldType = () => {
+      setIsExpressionField(!isExpressionField);
+    };
+
+    const onChangeSelectValue = (event, value) => {
+      props.onChange(event.target.value);
+    };
+
+    const onChangeTextValue = (value: string) => {
```

Comment:
Is it worth using callbacks for this method and the one above? I'm not sure

Reply:
probably yes, as they're passed down. I'll do that